File: Escola/Cursos.Repository/Services/AvaliacaoService.cs
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/11/2017 9:23:24 PM

///// Para outros metodos implementar classe partial
///// </sumary>
using System;
using System.Linq;
using System.Collections.Generic;
using System.Threading.Tasks;
using Cursos.Repository.Services.Base;
using Cursos.Data;
using Cursos.Repository.Services.Interfaces;
using System.Linq.Expressions;
using Cursos.Repository.Work.Interface;
namespace  Cursos.Repository.Services
{
	  /// <summary>
      /// Public Service Class  AvaliacaoService
      /// </summary>
      public partial class AvaliacaoService : BaseServices, IAvaliacaoService
      {

		public AvaliacaoService(IUnitOfWork unitOfWork)
        {
			UnitOfWork = unitOfWork;
        }

		/// <summary>
        /// Add Range Entity EV_AVALIACAO
        /// </summary>
        /// <param name="entity"></param>
		public void Add(IEnumerable<EV_AVALIACAO> entity)
        {
            UnitOfWork.AvaliacaoRepository.Add(entity);
        }

		/// <summary>
        /// Add Entity EV_AVALIACAO
        /// </summary>
        /// <param name="entity"></param>
		public void Add(EV_AVALIACAO entity)
        {	
            UnitOfWork.AvaliacaoRepository.Add(entity);
        }
		/// <summary>
        /// Delete Entity EV_AVALIACAO
        /// </summary>
        /// <param name="entity"></param>
        public void Delete(EV_AVALIACAO entity)
        {
            UnitOfWork.AvaliacaoRepository.Delete(entity);
        }
		 /// <summary>
        /// Get All Entity EV_AVALIACAO
        /// </summary>
        /// <returns>IEnumerable EV_AVALIACAO</returns>
        public IEnumerable<EV_AVALIACAO> GetAll()
        {
			return UnitOfWork.AvaliacaoRepository.GetAll();
        }
	     
		/// <summary>
        /// Update Entity EV_AVALIACAO
        /// </summary>
        /// <param name="entity"></param>
        public void Update(EV_AVALIACAO entity)
        {
            UnitOfWork.AvaliacaoRepository.Edit(entity);
        }

		/// <summary>
        /// Commit Transaction
        /// </summary>
        /// <returns></returns>
        public int Commit()
        {
            return UnitOfWork.Commit();
        }
		/// <summary>
        /// Commit Async Transaction
        /// </summary>
		public Task<int> CommitAsync()
        {
            return UnitOfWork.CommitAsync();
        }
				/// <summary>
        /// Get Entity EV_AVALIACAO By Id
        /// </summary>
        /// <param name="id"></param>
        /// <returns>EV_AVALIACAO</returns>
		public EV_AVALIACAO GetById(int id)
        {
			return UnitOfWork.AvaliacaoRepository.FindBy(x => x.ID.Equals(id)).SingleOrDefault();
        }		
				
		public IEnumerable<EV_AVALIACAO> FindBy(Expression<Func<EV_AVALIACAO, bool>> predicate)
        {
            return UnitOfWork.AvaliacaoRepository.FindBy(predicate);
        }
      }
}

File: Escola/Cursos.Api/scripts/service/avaliacaoService.js
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/13/2017 1:58:38 PM
///// Gerado automaticamente para alteracoes dos metodos comum alterar o template
///// </sumary>
"use strict";
app.factory("AvaliacaoService", ["$http", "ngAuthSettings", function ($http, ngAuthSettings) {
        var service = {};
		var serviceBase = ngAuthSettings.apiServiceBaseUri;
        service.GetAll = function (callback) {
            $http.get(serviceBase + "Avaliacao/getall")
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.GetById = function (id, callback) {
            $http.get(serviceBase + "Avaliacao/getbyid/" + id)
               .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Post = function (Avaliacao, callback) {
            $http.post(serviceBase + "Avaliacao/post", Avaliacao)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Put = function (Avaliacao, callback) {
            $http.post(serviceBase + "Avaliacao/put", Avaliacao)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

		service.Delete = function (Avaliacao, callback) {
            $http.post(serviceBase + "Avaliacao/delete", Avaliacao)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        return service;
    }]);

